File: app/src/main/java/rusha/x/MainActivity.java
package rusha.x;

import android.os.Bundle;
import android.view.LayoutInflater;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;

import rusha.x.databinding.MainActivityBinding;

public class MainActivity extends AppCompatActivity {
    public AppComponent di;
    public MainActivityBinding binding;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LayoutInflater inflater = getLayoutInflater();
        binding = MainActivityBinding.inflate(inflater);
        setContentView(binding.getRoot());
        di = DaggerAppComponent.create();

        NavController navController = Navigation.findNavController(this, R.id.navHostFragment);
        NavigationUI.setupWithNavController(binding.bottomNavigationView, navController);
    }

    public static AppComponent di(Fragment fragment) {
        return ((MainActivity) fragment.requireActivity()).di;
    }
}

File: app/src/main/java/rusha/x/MainActivity.kt
package rusha.x

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import androidx.navigation.ui.NavigationUI
import kotlinx.android.synthetic.main.main_activity.*

class MainActivity : AppCompatActivity() {
    lateinit var di: AppComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        di = DaggerAppComponent.create()
        NavigationUI.setupWithNavController(bottomNavigationView, navHostFragment.findNavController())
    }
}

fun Fragment.di(): AppComponent {
    return (this.requireActivity() as MainActivity).di
}